server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -320,4 +338,34 @@ public DirectoryReader getDirectoryReader() {
         assert reader instanceof DirectoryReader : "expected an instance of DirectoryReader, got " + reader.getClass();
         return (DirectoryReader) reader;
     }
+
+    private static class MutableQueryTimeout implements QueryTimeout {
```

Comment:
Please don't implement QueryTimeout, I think this introduces more issues than it helps, since its contract is that `shouldExit` should return true once the timeout is hit, while we instead use runnables that throw exceptions.

Reply:
Replaced with our own iface.